headlamp-k8s/plugins
# Add cert-manager plugin

# Testing the Cert-Manager Plugin

## Prerequisites
- A Kubernetes cluster with cert-manager installed
  - If you need to install cert-manager, follow the [official installation guide](https://cert-manager.io/docs/installation/)
- Node.js and npm installed on your system

## Steps to Test

1. Clone the plugins repository:
   ```bash
   git clone https://github.com/headlamp-k8s/plugins.git
   ```

2. Switch to the cert-manager branch:
   ```bash
   git checkout cert-manager
   ```

3. Navigate to the cert-manager plugin directory:
   ```bash
   cd cert-manager
   ```

4. Install the required dependencies:
   ```bash
   npm install
   ```

5. Start the plugin in development mode:
   ```bash
   npm run start
   ```

6. Launch Headlamp. You should now see "Cert Manager" in the sidebar.
## Optional: Generate Cert-Manager Types
To test the plugin with sample cert-manager resources:

1. Navigate to the test-files directory:
   ```bash
   cd test-files
   ```

2. Apply the sample configurations to your cluster:
   ```bash
   kubectl apply -f clusterIssuer.yaml
   kubectl apply -f issuer.yaml
   kubectl apply -f certificate.yaml
   kubectl apply -f order.yaml
   kubectl apply -f app.yaml
   ```

This will create:
- A ClusterIssuer for Let's Encrypt staging
- An Issuer for Let's Encrypt staging
- Two Certificate resources (one using ClusterIssuer, one using Issuer)
- An Order resource
- Sample Nginx deployment with Ingress configurations

Demo

https://github.com/user-attachments/assets/3bd40e22-94bc-4624-b7be-d303e3688dd9
